src/coreclr/System.Private.CoreLib/src/System/Delegate.CoreCLR.cs

```diff
@@ -156,7 +156,7 @@ public override int GetHashCode()
 
         protected virtual MethodInfo GetMethodImpl()
         {
-            if ((_methodBase == null) || !(_methodBase is MethodInfo))
+            if (_methodBase is null or not MethodInfo)
```

Comment:
```suggestion
            if (_methodBase is MethodInfo methodInfo)
            {
                return methodInfo;
            }
```
+ unindent the rest. It should avoids redundant cast on the hot-path.